homeassistant/components/torque/sensor.py

```diff
@@ -88,7 +88,12 @@ def get(self, request):
                 names[pid] = data[key]
             elif is_unit:
                 pid = convert_pid(is_unit.group(1))
-                units[pid] = data[key]
+
+                temp_unit = data[key]
+                if "\\xC2\\xB0" in temp_unit:
+                    temp_unit = temp_unit.replace("\\xC2\\xB0", "°")
```

Comment:
Isn't it possible to solve this by applying the correct encoding at some point in the flow after receiving the request?

Reply:
To be honest, I am not a proficient python coder, and this was the only way i was able to correct the units. The posted url is a query string with these characters. Seeing as how this was left effectively broken, I tinkered until I found a working solution. I'm all ears if you have a cleaner solution.

Reply:
Can you check if the following would fix your issue:

```
units[pid] = data[key].decide("utf-8")
```